pages/en/contrib/style-guide-es6.md

```diff
@@ -0,0 +1,764 @@
+  {title: 'Title Z', subject: 'A'},
+  {title: 'Title M', subject: 'C'},
+  {title: 'Title A', subject: 'A'},
+  {title: 'Title B', subject: 'A'},
+  {title: 'Title C', subject: 'D'}];
+```
+
+### Line spacing
+
+In general, group related lines together (with a single empty line in between groups).
+
+```
+if (err) return done(err);
+
+const cat = new Cat();
+cat.eat();
+cat.meow();
+cat.sleep();
+
+return cat;
+```
+
+However, if the method is short (3-5 lines) then just group it all together.
+
+**Good**:
+
+```
+if (err) return done(err);
+expect(result).to...;
+done();
+```
+
+**Bad**:
+
+```
+if (err) return done(err);
+
+expect(result).to...;
+
+done();
+```
+
+## Style guidelines for tests
+
+### Sandbox directories
+
+- All test-related sandbox directories should be inside the `test` dir (ie.
+  `./test/sandbox`)
+- Do not use directories like `/tmp/sandbox` as you will run into
+  permission issues on CI for directories that are not in the project
+
+### Email examples
+
+- All test-related email examples should be of the format `email@example.com`.
+- The `example.com` domain was created to be used for examples in documents, and could be used without prior coordination or asking for permission.
+
+**Good**:
+
+```js
+const validCredentials = {email: `original@example.com`, password: 'bar'}
+```
+
+**Bad**:
+
+```js
+const validCredentials = {email: `updated@bar.com`, password: 'bar'}
+```
+
+### Hooks
+
+**Good**:
+
+```js
+beforeEach(namedFunction);
+
+beforeEach(function namedFunction() {
+  // ...
+});
+```
+
+Shows up in stack traces, but not test output. In the second style, it's better to move the named function to the bottom of the file and call it using the first style instead.
+
+```js
+beforeEach('some description', function() {
+});
+
+beforeEach('some description', namedFunction);
+
+beforeEach('some description', function namedFunction() {
+  // ...
+});
+```
+
+The first example shows up in test output, but not stack traces. The second and third example shows up in test output and stack traces, but is a bit redundant to type two descriptions (one in the string and a duplicate in the function name)
+
+{% include tip.html content="Each of the above styles is acceptable and a decision will be made in the future as to which one is preferred. For now, feel free to pick the one that suits you.
+" %}
+
+**Bad**:
+
+```js
+beforeEach(function() {
+  ...
+});
+```
+
+### Layout of test files
+
+When using hooks like beforeEach/before, it's best to use named functions that are then defined at the bottom of the test file. The idea is to make it easy to find the meat of a test file, which are the unit-tests. The method names used for hooks should make it clear enough what's their purpose, allowing most readers to not need to know implementation details and skip directly to unit-tests.
+
+**Good**:
+
+```js
+describe('strong-error-handler', () => {
+  before(setupHttpServerAndClient);
+  beforeEach(resetRequestHandler)
```

Comment:
not sure if it matters.. is it supposed to have a trailing `;`

Reply:
Good catch!